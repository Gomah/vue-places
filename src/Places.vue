<template>
  <input type="search" v-bind:value="value" @input="updateValue($event.target.value)" />
</template>
<script>
import places from 'places.js';

export default {
  props: {
    value: {
      type: String,
      default: '',
    },
    options: {
      type: Object,
      default: () => ({}),
    },
  },

  data() {
    return {
      placesAutocomplete: null,
    };
  },

  methods: {
    updateValue(value) {
      this.$emit('input', value);
    },
  },

  mounted() {
    this.options.container = this.options.container || this.$el;
    this.placesAutocomplete = places(this.options);

    this.placesAutocomplete.on('change', (e) => {
      this.$emit('change', e.suggestion);
      this.updateValue(e.suggestion.value);
    });

    this.placesAutocomplete.on('clear', () => {
      this.updateValue(null);
    });
  },

  beforeDestroy() {
    this.placesAutocomplete.destroy();
  },
};
</script>
