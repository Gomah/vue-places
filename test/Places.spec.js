import { shallowMount } from '@vue/test-utils';
import Places from '../src/Places.vue';

describe('Places', () => {
  test('mounts properly', () => {
    const wrapper = shallowMount(Places);
    expect(wrapper.isVueInstance()).toBeTruthy();
  });

  test('renders properly', () => {
    const wrapper = shallowMount(Places);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
