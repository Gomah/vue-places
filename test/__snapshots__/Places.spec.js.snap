// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Places renders properly 1`] = `<input type="search" class="ap-input" autocomplete="off" spellcheck="false" role="combobox" aria-autocomplete="both" aria-expanded="false" aria-owns="algolia-places-listbox-1" style="vertical-align: top;" dir="auto">`;
